src/services/pathCompletions.ts

```diff
@@ -326,32 +328,40 @@ namespace ts.Completions.PathCompletions {
 
             if (typeRoots) {
                 for (const root of typeRoots) {
-                    getCompletionEntriesFromDirectories(host, root, span, result);
+                    getCompletionEntriesFromDirectories(root);
                 }
             }
-        }
 
-        if (host.getDirectories) {
             // Also get all @types typings installed in visible node_modules directories
             for (const packageJson of findPackageJsons(scriptPath, host)) {
                 const typesDir = combinePaths(getDirectoryPath(packageJson), "node_modules/@types");
-                getCompletionEntriesFromDirectories(host, typesDir, span, result);
+                getCompletionEntriesFromDirectories(typesDir);
             }
         }
 
         return result;
-    }
 
-    function getCompletionEntriesFromDirectories(host: LanguageServiceHost, directory: string, span: TextSpan, result: Push<CompletionEntry>) {
-        if (host.getDirectories && tryDirectoryExists(host, directory)) {
-            const directories = tryGetDirectories(host, directory);
-            if (directories) {
-                for (let typeDirectory of directories) {
-                    typeDirectory = normalizePath(typeDirectory);
-                    result.push(createCompletionEntryForModule(getBaseFileName(typeDirectory), ScriptElementKind.externalModuleName, span));
+        function getCompletionEntriesFromDirectories(directory: string) {
+            Debug.assert(!!host.getDirectories);

```

Comment:
This method is optional -- our test runner will always provide it though, so this assertion won't fail in any tests.

Reply:
I'm not sure exactly what you're suggestion (remove the assertion?) but I included this assertion because the code was extracted from under an `if (host.getDirectories)` check.  Possibly you're suggesting that that check isn't required either?

Reply:
Whoops, didn't notice that there's still an `if`, thought it was an assertion now.
Although it looks like `tryIOAndConsumeErrors` already handles the method being undefined, so the `if` may be unnecessary anyway.